src/raw_vec.rs

```diff
@@ -504,14 +504,12 @@ impl<T, A: AllocRef> RawVec<T, A> {
         } else {
             return Ok(());
         };
-        let new_size = amount * mem::size_of::<T>();
+        let new_layout = Layout::array::<T>(amount)?;
 
         let ptr = unsafe {
```

Comment:
The new layout can safely be constructed via `Layout::from_size_align_unchecked`; The new size is guaranteed to be smaller than or equal to the old size, the alignment is taken from the old layout.

```suggestion
        let ptr = unsafe {
            let new_size = amount * mem::size_of::<T>();
            let new_layout = Layout::from_size_align_unchecked(new_size, layout.align());
```